server/neptune/gateway/event/root_config_builder.go

```diff
@@ -67,7 +67,11 @@ func (b *RootConfigBuilder) Build(ctx context.Context, repo models.Repo, branch
 
 func (b *RootConfigBuilder) build(ctx context.Context, repo models.Repo, branch string, sha string, fileFetcherOptions github.FileFetcherOptions, installationToken int64) ([]*valid.MergedProjectCfg, error) {
 	// Generate a new filepath location and clone repo into it
-	repoDir, cleanup, err := b.RepoFetcher.Fetch(ctx, repo, branch, sha)
+	// TODO: consider supporting shallow cloning for comment based events too
+	repoFetcherOptions := github.RepoFetcherOptions{
```

Comment:
not a blocker but what do you think of passing `repoFetcherOption` to the root config builder so that it does not have to decide whether it is a shallow clone or not? 
PushEvent handler would pass in `repoFetcherOption` with shallow clone enabled and the PullEventHandlerProxy would pass in `repoFetcherOption` with shallow clone disabled for now.